src/settings.rs

```
@@ -482,6 +494,30 @@ impl ContextSettingExt for Context {
             Ok(None)
         }
     }
+    fn get_setting_for_contact(&self, person_id: PersonId) -> Option<(&dyn SettingType, usize)> {
```

Not for this PR, but as a longer-term discussion, I wonder what the use case is for this function returning None -- won't all people have some itinerary if the itinerary is what sets their setting membership?

---

Guido also asks if this should be an error rather than returning None. I think since this method is used for infection propagation, we should throw an error here. Presumably we expect everyone to have an itinerary. I will create this error here and it to the test for this method.

---

This is a little more involved than I first thought so I am going to put it is an issue